docs/tutorial/file_handling.md

```diff
@@ -0,0 +1,471 @@
+
+@:select(api-style)
+
+@:choice(syntax)
+
+```scala mdoc:compile-only
+import io.chrisdavenport.shellfish.syntax.path.*
+import fs2.io.file.Path
+import cats.effect.{IO, IOApp}
+
+object Creating extends IOApp.Simple:
+
+  val emptyDirectories = Path("create/me/first")
+
+  def run: IO[Unit] =
+    emptyDirectories.createDirectories >> Path("now_i_can_be_created.fs").createFile
+
+end Creating
+```
+
+@:choice(static)
+
+```scala mdoc:compile-only
+import io.chrisdavenport.shellfish.FilesOs
+import fs2.io.file.Path
+import cats.effect.{IO, IOApp}
+
+object Creating extends IOApp.Simple:
+
+  val emptyDirectories = Path("create/me/first")
+
+  def run: IO[Unit] =
+    FilesOs.createDirectories(emptyDirectories) >> 
+      FilesOs.createFile(Path("now_i_can_be_created.fs"))
+
+end Creating
+```
+
+@:@
+
+### Exercise
+
+Imagine that you need to create a lot of nested files in a directory that also needs the creation of the subsequent folders. For that reason, you need to implement a function that creates both the file and the parent directories. Can you implement it? How?
+
+@:select(api-style)
+
+@:choice(syntax)
+
+```scala
+extension (path: Path) def createFileAndDirectories: IO[Unit] = ???
+```
+
+@:choice(static)
+
+```scala
+def createFileAndDirectories(path: Path): IO[Unit] = ???
+```
+
+@:@
+
+[See possible implementation](https://gist.github.com/00c59408740e8b52838ff07fc9154c12.git)
+
+## Deleting here and there
+
+Creating is just the first part. Because memory is not infinite, you may also want to delete a file on your system.
+
+Here, you can use the `delete` method:
+
+@:select(api-style)
+
+@:choice(syntax)
+
+```scala mdoc:compile-only
+import io.chrisdavenport.shellfish.syntax.path.*
+import fs2.io.file.Path
+import cats.effect.{IO, IOApp}
+
+object Deleting extends IOApp.Simple:
+
+  val annoyingFile = Path("desktop/extend_your_car_warranty_now.ad")
+
+  def run: IO[Unit] =
+    for
+      exists <- annoyingFile.exists
+      _ <- if exists then annoyingFile.delete
+           else IO.unit
+    yield ()
+
+end Deleting
+```
+
+@:choice(static)
+
+```scala mdoc:compile-only
+import io.chrisdavenport.shellfish.FilesOs
+import fs2.io.file.Path
+import cats.effect.{IO, IOApp}
+
+object Deleting extends IOApp.Simple:
+
+  val annoyingFile = Path("desktop/extend_your_car_warranty_now.ad")
+
+  def run: IO[Unit] =
+    for
+      exists <- FilesOs.exists(annoyingFile)
+      _ <- if exists then FilesOs.delete(annoyingFile)
+           else IO.unit
+    yield ()
+
+end Deleting
+```
+
+@:@
+
+Note that we first check if the file exists before deleting it, this is because trying to delete a file with `delete` that does not exist will result in an error. If that is not your style, you have two options. One is using the `whenA` combinator from `cats.syntax.applicative.*`:
```

Comment:
```suggestion
Note that we are first checking if the file exists before deleting it, this is because trying to delete a file that does not exist will result in an error. To avoid this error, you have two options. One is using the [`whenA` combinator](https://github.com/typelevel/cats/blob/main/core/src/main/scala/cats/Applicative.scala#L263) from [Applicative](https://typelevel.org/cats/typeclasses/applicative.html) importing `cats.syntax.applicative.*`:
```